srtp.go

```diff
@@ -44,36 +44,63 @@ func (c *Context) DecryptRTP(dst, encrypted []byte, header *rtp.Header) ([]byte,
 	return c.decryptRTP(dst, encrypted, header)
 }
 
-func (c *Context) encryptRTP(dst, decrypted []byte, header *rtp.Header) ([]byte, error) {
-	dst = allocateIfMismatch(dst, decrypted)
+// EncryptRTP Encrypts a RTP packet
+// This method modifies rtp.Header based on the contents of plaintext.
+// TODO This function should take (*rtp.Packet) instead of ([]byte, rtp.Header)
+// TODO Don't modify *rtp.Header; ideally just remove it.
```

Comment:
This is on purpose. Users can pass in a `*rtp.Header` if they have already unmarshalled and we just modify it for them.

If they pass nil we allocate a new one

Reply:
I don't understand. The raw packet hasn't changed, so neither will `*rtp.Header` right? Maybe I'm missing the part of the code that modifies the packet.

It only seems useful if you have a raw packet and don't want to `Unmarshal` it until after you've sent it again... and you want to save nanoseconds.